File: 0072-edit-distance/0072-edit-distance.py
class Solution:
    def minDistance(self, text1: str, text2: str) -> int:
        n=len(text1)
        m=len(text2)
        
        dp = [[0 for _ in range(n+1)] for _ in range(m+1)]
        for i in range(n+1):
            dp[0][i]=i
        for i in range(m+1):
            dp[i][0]=i
        for i in range(1,n+1):
            for j in range(1,m+1):
                if text1[i-1]==text2[j-1]:
                    dp[j][i]=dp[j-1][i-1]
                else:
                    dp[j][i] = min(dp[j-1][i-1],dp[j][i-1],dp[j-1][i])+1
        return dp[-1][-1]
        
        

File: 0072-edit-distance/0072-edit-distance.cpp
#include <vector>
using namespace std;
class Solution {
public:
    int minDistance(string word1, string word2) {
        int n = word1.size();
        int m = word2.size();
        vector<vector<int>> dp(m+1, vector<int>(n+1,0));
        for(int i=0;i<=n;i++){
            dp[0][i]=i;
        }
        for (int j=0;j<=m;j++){
            dp[j][0]=j;
        }
        
        for(int j=1;j<=m;j++){
            for(int i=1;i<=n;i++){
                if(word1[i-1]==word2[j-1]){
                    dp[j][i]=dp[j-1][i-1];
                }else{
                    dp[j][i]=min({dp[j-1][i-1],dp[j][i-1],dp[j-1][i]})+1;
                }
            }
        }
        return dp[m][n];
    }
};